### DISCUSSION TITLE
Welcome to pyedflib Discussions!

### DISCUSSION THREAD

**Discussion by holgern** [holgern/pyedflib#98] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mishashevelinn** [comment]
Dear pyedflib developers and users,
I'm a student working in a medical startup.
I use pyEDFlib to parse EDF files to numpy for further analysis.
I wonder if there is some way to parse a subset of channels?
Currently, I'm parsing the whole file and then dropping some channels sometimes.
The thought is that predefinition of channels to be ignored while parsing may speed up the whole process. Is it possible?
Thank you.

**Reply by skjerns** [reply]
have a look at the highlevel interface

```Python
pyedflib.highlevel.read_edf(edf_file, ch_names=['ECG I'])
```

**Reply by skjerns** [reply]
if you want to ignore a subset of channels have a look at the source of `read_edf` and adapt it accordingly. you can do so easily, just look at the docs. currently there is no function to ignore certain channels, but feel free to make a PR to add it, should be rather simple to add.

**Comment by ataro1002** [comment]
I'm experiencing problems with Japanese characters in EDF+ annotations and  I've found two issues:

1. The `EdfReader._convert_string` method incorrectly prioritizes Latin-1 encoding over UTF-8. I've found this modification works better. 
```python
def _convert_string(self, s):
    if isinstance(s, bytes):
        try:
            return s.decode("utf_8", "strict")  # Prioritize UTF-8
        except UnicodeDecodeError:
            return s.decode("latin1", errors="replace")  # Fallback
    else:
        return s
```

2. When I make EDF file with annotations that contains Japanese characters. Japanese characters convert to ".".The C implementation of the edfwrite_annotation_utf8 function does not appear to be working well. 
I don't have the ability to fix the C code myself, but this is a significant limitation for using EDF+ with non-Latin languages.

It would be very helpful if you could fix these.
Thank you.

**Reply by skjerns** [reply]
this is probably related to https://github.com/holgern/pyedflib/issues/275

Need to check if the `convert_string` method still makes sense with python3 and default utf8. Afaik it's also a problem with cross-platform-compatibility and the C library, as we had to make this patch: https://github.com/holgern/pyedflib/blob/master/setup.py#L180

**Reply by skjerns** [reply]
I've implemented your fix here https://github.com/holgern/pyedflib/pull/276

will have a look at reading UTF8 files later

**Reply by skjerns** [reply]
fixed in #277
